### PR TITLE
2064 templates pages

### LINKED ISSUE #2064: Clean up templates/pages
## Type
- Enhancement

## Problem description
Lots of unnecessary pages.

## Expected behavior / Proposed solutions
Proposed pages:
- Home
- Modules
  - Widgets page with all widgets.
  - Page per action.
- Contact

Header enhancement:
- change languages switcher to bootstrap dropdown

### PR DESCRIPTION
**Pull request by jonasdekeukelaere**
## Type
- Enhancement

## Resolves the following issues
Fixes #2064 

## Pull request description
Cleans up pages when installing with example data
screen shot 2017-08-01 at 10 01 01
screen shot 2017-08-01 at 10 02 06

### CODE REVIEW

File: src/Backend/Modules/Pages/Installer/Installer.php [hunk 1]
```diff
@@ -65,14 +65,147 @@ private function installExampleData(): void
...
+            'Logout',
+            'Logout'
+        );
+        $extras['profiles_change_email'] = $this->insertExtra(
+            'Profiles',
+            ModuleExtraType::block(),
+            'ChangeEmail',
+            'ChangeEmail'
+        );
+        $extras['profiles_change_password'] = $this->insertExtra(
```
Comment (carakas) [inline src/Backend/Modules/Pages/Installer/Installer.php]: What happens if you install fork for instance without the profiles module?
Won't this then also try to add the profile widgets and blocks?

Comment (jonasdekeukelaere) [inline src/Backend/Modules/Pages/Installer/Installer.php]: Hmm probably, is there a way to get the selected modules from the installer in the install method?

Comment (carakas) [inline src/Backend/Modules/Pages/Installer/Installer.php]: don't know that by heart, you'll have to look for it, or add it if needed.
But wouldn't it be better to just add these things in the installers of the modules?

Comment (jonasdekeukelaere) [inline src/Backend/Modules/Pages/Installer/Installer.php]: Fixed 0cd3bfe

File: src/Backend/Modules/Mailmotor/Installer/Installer.php [hunk 2]
```diff
@@ -61,9 +61,12 @@ private function configureFrontendPages(): void
     {
         // loop languages
         foreach ($this->getLanguages() as $language) {
-            $pageId = $this->insertPage(
-                ['title' => 'Newsletters', 'language' => $language]
-            );
+            $pageId = $this->getPageWithMailmotorBlock($language);
+            if ($pageId === null) {
+                $pageId = $this->insertPage(
+                    ['title' => 'Newsletters', 'language' => $language]
```
Comment (tijsverkoyen) [inline src/Backend/Modules/Mailmotor/Installer/Installer.php]: Why split all parameters to new lines, but leave the full array on a single line?

File: src/Backend/Modules/Pages/Installer/Installer.php [hunk 3]
```diff
@@ -65,14 +65,153 @@ private function installExampleData(): void
...
+                'SecurePage'
+            );
+        }
 
         // loop languages
         foreach ($this->getLanguages() as $language) {
+            // insert modules page
+            $this->insertPage(
+                [
+                    'id' => 4,
```
Comment (tijsverkoyen) [inline src/Backend/Modules/Pages/Installer/Installer.php]: Wouldn't it be more logical to store this ID in a variable?

File: src/Backend/Modules/Pages/Installer/Installer.php [hunk 4]
```diff
@@ -100,6 +239,7 @@ private function installExampleData(): void
                             $this->getLocale('Blog', 'Core', $language, 'lbl', 'Frontend')
                         ),
                         'language' => $language,
+                        'parent_id' => 4,
```
Comment (tijsverkoyen) [inline src/Backend/Modules/Pages/Installer/Installer.php]: See above, use a variable instead of a hardcoded value.

Comment (jonasdekeukelaere) [inline src/Backend/Modules/Pages/Installer/Installer.php]: Fixed 7255e29bc

File: src/Backend/Modules/Pages/Installer/Installer.php (hunk 3 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Pages/Installer/Installer.php]: Fixed 7255e29bc

File: src/Backend/Modules/Mailmotor/Installer/Installer.php (hunk 2 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Mailmotor/Installer/Installer.php]: I copied it from what was already there. Just added an if.

Comment (tijsverkoyen) [inline src/Backend/Modules/Mailmotor/Installer/Installer.php]: 👍

**Pull request merged**